src/ArmTemplates/Commands/Executors/ExtractorExecutor.cs

```diff
@@ -45,7 +45,7 @@ public class ExtractorExecutor
             ILoggerExtractor loggerExtractor,
             IMasterTemplateExtractor masterTemplateExtractor,
             IPolicyExtractor policyExtractor,
-            IProductApiExtractor productApiExtractor,
+            IServiceApisProductsExtractor productApisExtractor,
```

Comment:
I was a little bit confused by the namings - because `ProductApiExtractor` was really getting information about products, not about the APIs connected to APIM instance. And I tried to rename it in a way it is a little bit more understandable. In example:
- `IServiceApisApiClient` - retrieves service APIs (aka APIM instance apis)
- `IServiceApiProductsApiClient` - retrieves products of service APIs 

If you think it is a bad initiative, please, help me with choosing proper naming. It is hard enough :D

Reply:
Why not just 
`IServiceApiClient`
`IServiceProductsClient`

Or even 

`IApisClient`
`IProductsClient`

Reply:
Another example: 
`ServiceApisProductTemplateResource` which contains `ServiceApiProductProperties` could just become `ProductTemplateResource` and contain "ProductProperties".

Reply:
I agree that we can just simplify this to `IServiceClient` and `IProductsClient`.

Reply:
Maybe then `IApisClient` and `IProductsClient`? Because in `IServiceClient` we are only getting APIM APIs data

Reply:
I am sorry, as I found out we have also `ProductExtractor`. So we really need `ProductApisExtractor` and `ProductExtractor` and we can do nothing about that.

I will now rename everything properly and we will see if it is good for you

Reply:
Yes, it is the following: `ProductApisExtractor` is getting info about APIs of products, but `ProductExtractor` just retrieves info about products. It is very confusing for me also, because I'm not really understanding purposes of some code. Sorry for that.

@Hardell @tomkerkhove thanks for your comments, I've done global renaming for this PR - please review it once more